src/cascadia/TerminalApp/ColorScheme.cpp

```diff
@@ -1,222 +1,237 @@
-    if (auto table{ json[JsonKey(TableKey)] })
-    {
-        int i = 0;
-
-        for (const auto& tableEntry : table)
-        {
-            if (tableEntry.isString())
-            {
-                auto color = Utils::ColorFromHexString(tableEntry.asString());
-                _table.at(i) = color;
-            }
-            i++;
-        }
-    }
-
-    int i = 0;
-    for (const auto& current : TableColors)
-    {
-        if (auto str{ json[JsonKey(current)] })
-        {
-            const auto color = Utils::ColorFromHexString(str.asString());
-            _table.at(i) = color;
-        }
-        i++;
-    }
-}
-
-std::wstring_view ColorScheme::GetName() const noexcept
-{
-    return { _schemeName };
-}
-
-std::array<COLORREF, COLOR_TABLE_SIZE>& ColorScheme::GetTable() noexcept
-{
-    return _table;
-}
-
-COLORREF ColorScheme::GetForeground() const noexcept
-{
-    return _defaultForeground;
-}
-
-COLORREF ColorScheme::GetBackground() const noexcept
-{
-    return _defaultBackground;
-}
-
-// Method Description:
-// - Parse the name from the JSON representation of a ColorScheme.
-// Arguments:
-// - json: an object which should be a serialization of a ColorScheme object.
-// Return Value:
-// - the name of the color scheme represented by `json` as a std::wstring optional
-//   i.e. the value of the `name` property.
-// - returns std::nullopt if `json` doesn't have the `name` property
-std::optional<std::wstring> TerminalApp::ColorScheme::GetNameFromJson(const Json::Value& json)
-{
-    if (const auto name{ json[JsonKey(NameKey)] })
-    {
-        return GetWstringFromJson(name);
-    }
-    return std::nullopt;
-}
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ColorScheme.h"
+#include "../../types/inc/Utils.hpp"
+#include "Utils.h"
+#include "JsonUtils.h"
+
+using namespace ::Microsoft::Console;
+using namespace TerminalApp;
+using namespace winrt::Microsoft::Terminal::Settings;
+using namespace winrt::Microsoft::Terminal::TerminalControl;
+
+static constexpr std::string_view NameKey{ "name" };
+static constexpr std::string_view TableKey{ "colors" };
+static constexpr std::string_view ForegroundKey{ "foreground" };
+static constexpr std::string_view BackgroundKey{ "background" };
+static constexpr std::string_view SelectionBackgroundKey{ "selectionBackground" };
+static constexpr std::array<std::string_view, 16> TableColors = {
+    "black",
+    "red",
+    "green",
+    "yellow",
+    "blue",
+    "purple",
+    "cyan",
+    "white",
+    "brightBlack",
+    "brightRed",
+    "brightGreen",
+    "brightYellow",
+    "brightBlue",
+    "brightPurple",
+    "brightCyan",
+    "brightWhite"
+};
+
+ColorScheme::ColorScheme() :
+    _schemeName{ L"" },
+    _table{},
+    _defaultForeground{ RGB(242, 242, 242) },
+    _defaultBackground{ RGB(12, 12, 12) },
+    _selectionBackground{ RGB(242, 242, 242) }
```

Comment:
Should this (and the others) be `DEFAULT_FOREGROUND` (etc.) from `DefaultSettings.h`?

Reply:
Yup, you're right, I've mimicked the default foreground and background from what they're set as in `TerminalSettings.cpp`, and changed selectionBackground to use `DEFAULT_FOREGROUND`.